test/e2e/testdata/workflow-terminate-on-exit-handler.yaml

```
@@ -0,0 +1,47 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: workflow-terminate-on-exit-handler
+  namespace: argo
+  annotations:
+    description: 'Test workflow termination with exit handler'
+spec:
+  podGC:
+    strategy: OnPodCompletion
+  entrypoint: workflow-startup
+  onExit: exit-handler
+  templates:
+  - name: workflow-startup
+    dag:
+      tasks:
+      - name: a
+        template: echo-template
+        arguments:
+          parameters:
+            - name: message
+              value: Hi!
+  - name: sleep
+    inputs:
+      parameters:
+        - name: time-value
+    container:
+      image: alpine:3.19.1
+      command: [sh, -c]
+      args: ["sleep {{inputs.parameters.time-value}}"]
+  - name: echo-template
+    inputs:
+      parameters:
+      - name: message
+    script:
+      image: python:alpine3.6
+      command: [python]
+      source: |
+        print("{{inputs.parameters.message}}")
+  - name: exit-handler
+    steps:
+      - - arguments:
+            parameters:
+              - name: time-value
+                value: "600"
+          name: exit-handler-task
+          template: sleep
```

```suggestion
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  name: workflow-exit-handler-sleep
spec:
  entrypoint: argosay-template
  onExit: exit-handler
  templates:
  - name: argosay-template
    container:
      image: argoproj/argosay:v2
      args: ["echo", "hello-world"]
  - name: sleep
    container:
      image: argoproj/argosay:v2
      args: ["sleep", "600"]
  - name: exit-handler
    steps:
      - - name: exit-handler-task
          template: sleep
```
- let's try keeping the test data simpler, easier to read.
- rename the file, since it's in testdata, I think it should name according to what's in the workflow not how you will action on the workflow, e.g. shouldn't have `terminate` in the name since the workflow doesn't terminate
- have the file in test/e2e/functional, since it is called by the functional_test

---

Moved it and applied your suggestions